sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -490,15 +510,17 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
             case c @ Cast(ne: NamedExpression, _, _, _) => Alias(c, ne.name)()
             case e: ExtractValue =>
```

Comment:
to match multipart identifier, we need to match a chain of `GetStructField` or `GetArrayStructField`